.github/workflows/azure-podvm-image.yml

```diff
@@ -0,0 +1,122 @@
+name: azure-podvm-image
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  PODVM_IMAGE_NAME: "peerpod-image-${{ github.run_id }}-${{ github.run_attempt }}"
+  SSH_USERNAME: "peerpod"
+  # VM size used for building image.
+  VM_SIZE: "Standard_D2as_v5"
+
+on:
+  schedule:
+    # Run at 12:00 AM UTC
+    - cron: '0 0 * * *'
+
+jobs:
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    outputs:
+      pod-image-version: "${{ steps.generate_image_version.outputs.pod_image_version }}"
+    steps:
+    - name: Generate version for pod vm image
+      id: generate_image_version
+      run: |
+        unique_version=$(date +'%Y.%m.%d')
+        echo "Generated unique version for the image as: ${unique_version}"
+        echo "pod_image_version=${unique_version}" >> $GITHUB_OUTPUT
+
+    - uses: actions/checkout@v3
+      with:
+        path: cloud-api-adaptor
+
+    - name: Clone kata repository
+      uses: actions/checkout@v3
+      with:
+        repository: kata-containers/kata-containers 
+        path: kata-containers
+        ref: CC-0.7.0
+
+    - uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: 1.69.0
+        default: true
+
+    - name: Set up rust build cache
+      uses: actions/cache@v3
+      continue-on-error: false
+      with:
+        # The paths to cache are documented here: https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci
+        path: |
+          ~/.cargo/.crates.toml
+          ~/.cargo/.crates2.json
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+        key: rust
```

Comment:
I think this cache is worth it, because it saves a few double digit seconds every time those files are downloaded. And yes, we do hit this cache because they key is static. Also those recommendations are from rust upstream: 
<img width="2127" alt="image" src="https://github.com/confidential-containers/cloud-api-adaptor/assets/5815795/7dca9dd1-8af6-4109-84d3-0f5e9fa25ef3">